tests_and_analysis/test/euphonic_test/test_force_constants_calculate_qpt_phonons_castep.py

```diff
@@ -0,0 +1,492 @@
+                 6.1104623404e-03, 6.3230407709e-03, 6.3262959354e-03],
+                [7.6869781440e-04, 8.8171407111e-04, 2.1005260482e-03,
+                 2.1043510112e-03, 3.5723508344e-03, 3.5925812023e-03,
+                 3.7679565852e-03, 3.7804526859e-03, 6.6594311404e-03,
+                 6.6651464171e-03, 7.2007012512e-03, 7.2132353897e-03],
+                [1.4065426444e-04, 1.4065426444e-04, 1.4065426444e-04,
+                 1.4065426444e-04, 3.2305904460e-04, 3.2305904460e-04,
+                 4.0222682768e-03, 4.0222682768e-03, 7.1591503492e-03,
+                 7.1591503492e-03, 7.1591503492e-03, 7.1591503492e-03]
+            ]) * ureg('hartree')
+        }
+    }
+
+    materials = [
+        (
+            "LZO", 'La2Zr2O7.castep_bin',
+            np.array([  # Qpoints
+                [-1.00, 9.35, 3.35],
+                [-1.00, 9.00, 3.00]
+            ])
+        ),
+        (
+            "graphite", "graphite.castep_bin",
+            np.array([  # Qpoints
+                [0.00, 0.00, 0.00],
+                [0.001949, 0.001949, 0.00],
+                [0.50, 0.00, 0.00],
+                [0.25, 0.00, 0.00],
+                [0.00, 0.00, 0.50]
+            ])
+        )
+    ]
+
+    kwargs = [
+        {"use_c": False, "fall_back_on_python": True, "n_threads": 1},
+        {"use_c": True, "fall_back_on_python": False, "n_threads": 1},
+        {"use_c": True, "fall_back_on_python": False, "n_threads": 2},
+        {
+            "use_c": False, "fall_back_on_python": True,
+            "n_threads": 1, "asr": "realspace"
+        },
+        {
+            "use_c": True, "fall_back_on_python": False,
+            "n_threads": 1, "asr": "realspace"
+        },
+        {
+            "use_c": True, "fall_back_on_python": False,
+            "n_threads": 2, "asr": "realspace"
+        }
+    ]
+
+    @pytest.fixture(
+        params=list(itertools.product(materials, kwargs))
+    )
+    def calculate_qpoint_phonon_modes(self, request):
+        material, kwargs = request.param
+        material_name, castep_bin_file, qpts = material
+        filename = os.path.join(self.path, material_name, castep_bin_file)
+        return ForceConstants.from_castep(filename), qpts, material_name, kwargs
+
+    def test_calculate_qpoint_phonon_modes(self, calculate_qpoint_phonon_modes):
+        fc, qpts, material_name, kwargs = calculate_qpoint_phonon_modes
+        print(material_name)
+        qpoint_phonon_modes = fc.calculate_qpoint_phonon_modes(
+            qpts, **kwargs
+        )
+        if "asr" in kwargs:
+            test_expected_freqs = self.expected_freqs[material_name]["asr"]
+        else:
+            test_expected_freqs = self.expected_freqs[material_name]["no_asr"]
+        npt.assert_allclose(
+            qpoint_phonon_modes.frequencies.to('hartree').magnitude,
+            test_expected_freqs.to('hartree').magnitude,
+            atol=1e-10
+        )
```

Comment:
Can we parametrize `calculate_qpoint_phonon_modes` so that we put in the expected frequencies as a parameter rather than having to use `if "asr" in kwargs` etc.? `test_cell_volume` in `test_crystal` might be a good template for this. This would also allow us to read from a .json file in all cases.

Reply:
Refactored to be a parameter which selects the expected frequencies by giving the material name and a key to identify the expected frequencies e.g realspace or no_asr